lib/elixir/src/elixir_expand.erl

```diff
@@ -817,12 +817,17 @@ expand_remote(Receiver, DotMeta, Right, Meta, Args, #{context := Context} = E, E
 
   {EArgs, {EA, _}} = lists:mapfoldl(fun expand_arg/2, {EL, E}, Args),
 
-  case rewrite(Context, Receiver, AttachedDotMeta, Right, Meta, EArgs) of
-    {ok, Rewritten} ->
-      maybe_warn_comparison(Rewritten, Args, E),
-      {Rewritten, elixir_env:close_write(EA, E)};
-    {error, Error} ->
-      form_error(Meta, E, elixir_rewrite, Error)
+  case lists:keyfind(no_parens, 1, Meta) of
+    {no_parens, true} when Context == guard andalso is_tuple(Receiver) ->
+      {{{'.', DotMeta, [Receiver, Right]}, Meta, EArgs}, EA};
```

Comment:
Not sure if we should also do `elixir_env:close_write(EA, E)` here?

Reply:
You should. My suggestion is to translate all of the arguments and build the new E without caring about the Context and only at the end you check the `Context` and the meta. In particular, I think we should raise a clear error message if we are in a guard and the receiver is an atom or there are parens.

Reply:
I pushed a new commit with a specific error message for `map.field()`, the `invalid_guard` error message in `elixir_rewrite` should handle the case when receiver is an atom.

We don't have to translate args since they are always empty when `no_parens: true`, because of this I don't think we have to call `elixir_env:close_write/2` either.